File: LeetCode/Easy/Find Most Frequent Vowel and Consonant/3541.java
class Solution {
    public int maxFreqSum(String s) {
        int[] arr = new int[26];
        for (char c : s.toCharArray()) arr[c - 'a']++;

        int maxVowel = 0, maxConsonant = 0;
        for (int i = 0; i < 26; i++) {
            char c = (char) ('a' + i);
            if (isVowel(c)) {
                maxVowel = Math.max(maxVowel, arr[i]);
            } else {
                maxConsonant = Math.max(maxConsonant, arr[i]);
            }
        }

        return maxVowel + maxConsonant;
    }

    public boolean isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
}

File: LeetCode/Easy/Find Most Frequent Vowel and Consonant/3541.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function(s) {
    const vowel = new Set(['a', 'e', 'i', 'o', 'u']);
    const arr = new Int32Array(26);
    for (let i = 0; i < s.length; i++) arr[s.charCodeAt(i) - 97]++;

    let maxVowel = 0, maxConsonant = 0;
    for (let i = 0; i < 26; i++) {
        const c = String.fromCharCode(i + 97);
        if (vowel.has(c)) {
            maxVowel = Math.max(maxVowel, arr[i]);
        } else {
            maxConsonant = Math.max(maxConsonant, arr[i]);
        }
    }

    return maxVowel + maxConsonant;
};
